src/app/components/FriendsList/FriendsList.tsx

```diff
@@ -11,43 +12,103 @@ import {
   DialogTitle,
 } from "@/components/ui/dialog";
 import FriendsListForm from "@/app/components/FriendsListForm/FriendsListForm";
+import { FriendItem } from "./FriendItem";
+import { IncomingRequestItem } from "./IncomingRequestItem";
+import { SentRequestItem } from "./SentRequestItem";
+import { api } from "@/trpc/client";
 
-interface Friend {
-  id: string;
-  email: string;
-  name?: string;
-}
-
-// Mocked friends list
-const MOCKED_FRIENDS: Friend[] = [
-  { id: "1", email: "alice@example.com", name: "Alice" },
-  { id: "2", email: "bob@example.com", name: "Bob" },
-  { id: "3", email: "charlie@example.com", name: "Charlie" },
-  { id: "4", email: "dave@example.com", name: "Dave" },
-  { id: "5", email: "eve@example.com", name: "Eve" },
-  { id: "6", email: "frank@example.com", name: "Frank" },
-  { id: "7", email: "george@example.com", name: "George" },
-  { id: "8", email: "hannah@example.com", name: "Hannah" },
-  { id: "9", email: "ian@example.com", name: "Ian" },
-  { id: "10", email: "jane@example.com", name: "Jane" },
-  { id: "11", email: "karen@example.com", name: "Karen" },
-  { id: "12", email: "larry@example.com", name: "Larry" },
-  { id: "13", email: "mary@example.com", name: "Mary" },
-  { id: "14", email: "nathan@example.com", name: "Nathan" },
-  { id: "15", email: "olivia@example.com", name: "Olivia" },
-  { id: "16", email: "patrick@example.com", name: "Patrick" },
-  { id: "17", email: "quinn@example.com", name: "Quinn" },
-];
+type TabValue = "friends" | "incoming" | "sent";
 
 function FriendsList() {
   const { t } = useTranslation();
   const [isOpen, setIsOpen] = useState(false);
-  const friends = MOCKED_FRIENDS;
+  const [activeTab, setActiveTab] = useState<TabValue>("friends");
+
+  const { data: friends = [], isLoading: friendsLoading } = api.friendships.getFriends.useQuery();
+  const { data: pendingRequests = [], isLoading: pendingLoading } = api.friendships.getPendingRequests.useQuery();
+  const { data: sentRequests = [], isLoading: sentLoading } = api.friendships.getSentRequests.useQuery();
 
   const onOpenModalHandler = () => {
     setIsOpen(true);
   };
 
+  const isLoading = friendsLoading || pendingLoading || sentLoading;
```

Comment:
I would simplify this component this way:

- we can use this component: https://ui.shadcn.com/docs/components/tabs
- i would split each tab for its own component

this one my personal preference
type TabValue = "friends" | "incoming" | "sent"; =>
```
export type ValueOf<T> = T[keyof T];

const TAB_VALUES = {
  FRIENDS: "friends",
  INCOMING: "incoming",
  SENT: "sent"
} as const;

type TabValuesType = ValueOf<typeof TAB_VALUES>
```

I also added loader, so you can show it if necessary